round4/liuxk.md

```diff
@@ -0,0 +1,50 @@
+
+## Target Audience
+
+Targeted towards visually impaired individuals, specifically for colorblind and near-blind. Our extension will adjust the style of a website so that any text is larger. In addition, any coloring/style will be adjusted so that it's more clearly visible for individuals with color blindness.
+Many individuals that are color blind or visually impaired struggle with viewing web pages that are not formatted well. They would significantly benefit from a tool that enlarges text and changes colors to be color-blind friendly. 
+Additionally, the extension can also be used by other people who simply want an easy way to change such features for their own personal preferences.
+
+## Description
+
+This extension allows users to change visual features (e.g., text size, color) on websites to suit their preferences. Giving users a way to interact with these visual elements more directly and easily, the tool can be used for a variety of purposes, from accessibility to personalization.
+
+## Selling Points
+
+1. Enhanced accessibility for everyone.
+2. Customizable styling. Users can change the size of fonts as well as the color/filters applied.
+3. Convenience. Chrome extension will be applied automatically to every webpage.
+4. Compatibility. Will be compatible with any type of website.
+5. Personalization. Users can create and save certain themes/styles as favorites.
+
+## User Stories
+
+_[List user stories that describe the main features of your Chrome Extension. Use the following template: "As a [user role], I want to [goal] so that [benefit]." Fill in the brackets with the appropriate information. For Round 1, write 5 user stories; Round 2, write 10 user stories; Round 3, write 15 user stories.]_
+1. As a colorblind user, I want to change the colors of various visual elements to colors that I can easily distinguish.
+2. As a visually impaired user, I want to increase text size to make the text more readable.
+3. As a user, I want to personalize webpages by changing colors, fonts, etc., to ones that I like.
+4. As a user, I want to create certain styles that I like and save them as “favorites,” allowing me to easily apply them to other sites in the future.
+5. As a user, I want to change background color to eliminate any distractions in the background.
+6. As a user, I want to access some built-in themes, like dark mode and grayscale.
+
+*Suggestion* color picker (like the eyedrop looking thing in paint softwares) that distinguishes very basic colors like red, yellow, blue and tells them to the user
+
+## Notes
+
+Color models in computer graphics
+Consider using HSL or HSV to determine hue (red from yellow from blue, etc)
+See basics of color section and “determine the value ranges [of the HSV or HSB model implementation used] before attempting to interpret a value.”
+Rule of thumb: Start simple and reliable. Better to have a few features that always work instead many features that sometimes work.
```

Comment:
@liuxk83 Indeed, you should try to prototype the style replacement feature early: I recommend you try to find out how to make small modifications in the **JavaScript console** to learn how to modify styling dynamically.

For instance, here is a conversation with Claude, where I am learning how to change the color red:
<img width="1264" alt="Screenshot 2024-04-02 at 2 50 10 PM" src="https://github.com/CIS-3500/ideation/assets/11770469/447da0bf-48c7-4a21-adb4-1256d9e826aa">

The better code to inject here though is:
```javascript
const style = document.createElement('style');
style.innerHTML = `
  * {
    color: red !important;
  }
`;
document.head.appendChild(style);
```
See Claude conversation for more information: [colorizing DOM text in red.pdf](https://github.com/CIS-3500/ideation/files/14842526/colorizing.DOM.text.in.red.pdf)